app/controllers/dashboard/calendars_controller.rb

```diff
@@ -20,9 +20,10 @@ def show
     end
 
     def day
-      date = params[:date].to_datetime
-      start_date = date.utc
-      end_date = date.end_of_day.utc
+      date = params[:date].to_datetime.utc
+      user_offset = ActiveSupport::TimeZone[current_user.time_zone].utc_offset
```

Comment:
Why is it needed? Every action has user's time zone already set in application_controller